addons/stock/models/stock_move.py

```diff
@@ -652,7 +658,11 @@ def action_assign_serial(self):
     def action_product_forecast_report(self):
         self.ensure_one()
         action = self.product_id.action_product_forecast_report()
-        action['context'] = {'warehouse': (self.picking_type_id or self.picking_id.picking_type_id).warehouse_id.id,}
+        warehouse = self.location_id.get_warehouse()
+        if warehouse:
+            action['context'] = {'warehouse': warehouse.id, }
+        else:
+            action['context'] = {}
```

Comment:
```suggestion
        action['context'] = {'warehouse': warehouse.id, } if warehouse else {}
```
It's only a proposition to make it shorter but keep the one you prefer.